network-area-diagram/src/main/java/com/powsybl/nad/build/iidm/VoltageLevelFilter.java

```diff
@@ -99,6 +101,64 @@ private static void traverseVoltageLevels(Set<VoltageLevel> voltageLevelsDepth,
         traverseVoltageLevels(nextDepthVoltageLevels, depth - 1, visitedVoltageLevels);
     }
 
+    private static void traverseVoltageLevelsWithVoltageFilter(Set<VoltageLevel> voltageLevelsDepth, int depth, Set<VoltageLevel> visitedVoltageLevels,
+                                              double highNominalVoltageBound, double lowNominalVoltageBound) {
+        if (depth >= 0) {
+            Set<VoltageLevel> nextDepthVoltageLevels = new HashSet<>();
+            for (VoltageLevel vl : voltageLevelsDepth) {
+                if (!visitedVoltageLevels.contains(vl)) {
+                    if (highNominalVoltageBound > 0 && lowNominalVoltageBound > 0) {
+                        if (vl.getNominalV() >= lowNominalVoltageBound
+                                && vl.getNominalV() <= highNominalVoltageBound) {
+                            traverseVoltageLevel(visitedVoltageLevels, nextDepthVoltageLevels, vl);
+                        }
+                    } else if (highNominalVoltageBound > 0) {
+                        if (vl.getNominalV() <= highNominalVoltageBound) {
+                            traverseVoltageLevel(visitedVoltageLevels, nextDepthVoltageLevels, vl);
+                        }
+                    } else if (lowNominalVoltageBound > 0) {
+                        if (vl.getNominalV() >= lowNominalVoltageBound) {
+                            traverseVoltageLevel(visitedVoltageLevels, nextDepthVoltageLevels, vl);
+                        }
+                    } else {
+                        traverseVoltageLevel(visitedVoltageLevels, nextDepthVoltageLevels, vl);
+                    }
+                }
+            }
+            traverseVoltageLevelsWithVoltageFilter(nextDepthVoltageLevels, depth - 1, visitedVoltageLevels, highNominalVoltageBound, lowNominalVoltageBound);
+        }
+    }
+
+    private static void traverseVoltageLevel(Set<VoltageLevel> visitedVoltageLevels, Set<VoltageLevel> nextDepthVoltageLevels, VoltageLevel vl) {
+        visitedVoltageLevels.add(vl);
+        vl.visitEquipments(new VlVisitor(nextDepthVoltageLevels, visitedVoltageLevels));
+    }
+
+    public static VoltageLevelFilter createNominalVoltageFilter(Network network, List<String> voltageLevelIds,
+                                                             double highNominalVoltageBound,
+                                                             double lowNominalVoltageBound, int depth) {
+        Objects.requireNonNull(network);
```

Comment:
But we could check here that the bounds are positive and that lowerBound is smaller or equal to the upperBound.